boot/seal_test.go

```diff
@@ -367,3 +375,65 @@ func (s *sealSuite) TestSealKeyModelParams(c *C) {
 				secboot.NewLoadChain(oldkbf))),
 	})
 }
+
+func (s *sealSuite) TestIsResealNeeded(c *C) {
+	chains := []boot.BootChain{
+		{
+			BrandID:        "mybrand",
+			Model:          "foo",
+			Grade:          "signed",
+			ModelSignKeyID: "my-key-id",
+			AssetChain: []boot.BootAsset{
+				// hashes will be sorted
+				{Role: bootloader.RoleRecovery, Name: "shim", Hashes: []string{"x", "y"}},
+				{Role: bootloader.RoleRecovery, Name: "loader", Hashes: []string{"c", "d"}},
+				{Role: bootloader.RoleRunMode, Name: "loader", Hashes: []string{"z", "x"}},
+			},
+			Kernel:         "pc-kernel-other",
+			KernelRevision: "2345",
+			KernelCmdlines: []string{`snapd_recovery_mode=run foo`},
+		}, {
+			BrandID:        "mybrand",
+			Model:          "foo",
+			Grade:          "dangerous",
```

Comment:
out of curiosity why is grade different for the recover chain vs the run chain?

Reply:
The data was copied from elsewhere, the idea was to introduce more variability when generating predictable boot chains.